source/tests/system/nifgen_driver_api_tests.cpp

```diff
@@ -62,7 +62,7 @@ class NiFgenDriverApiTest : public ::testing::Test {
     fgen::InitializeWithChannelsRequest request;
     request.set_channel_name("");
     request.set_resource_name("FakeDevice");
-    request.set_option_string("Simulate=1, DriverSetup=Model:5441;BoardType:PXI");
+    request.set_option_string("Simulate=1, DriverSetup=Model:5433;BoardType:PXI");
```

Comment:
Are you sure this works? 
- There are two models of 5433: "5433 (1CH)" and "5433 (2CH)". Specifying "Model:5433" without the # of channels returns `IVI_ERROR_RESOURCE_UNKNOWN` on my test machine.
- The correct board type is PXIe, but the driver seems to accept PXI.

Reply:
https://knowledge.ni.com/KnowledgeArticleDetails?id=kA03q000000xAKnCAM&l=en-US

Reply:
So I did end up having to switch to 5433 (2CH). Not sure if I maybe didn't rebuild correctly or ran the unit test runner instead accidently. 

The NiFgenSessionTests all run fine with the update to "5433 (2CH)". NiFgenDriverApi does not. Get the following test failures.  The first failure seems like a simple update but for the others I am not familiar enough with NIFGEN to determine a fix initially. I may remove the update to NiFgenDriverApiTest and create a new issue for its intermittency since it has more details to look through.
```batch
[ RUN      ] NiFgenDriverApiTest.PerformSelfTest_CompletesSuccessfuly
C:\Users\mboehme\github\forks\grpc-device\source\tests\system\nifgen_driver_api_tests.cpp(432): error: Expected equality of these values:
  ""
  response.self_test_message()
    Which is: "Self test passed"
[  FAILED  ] NiFgenDriverApiTest.PerformSelfTest_CompletesSuccessfuly (530 ms)
[ RUN      ] NiFgenDriverApiTest.SendSoftwareEdgeTrigger_TriggersSuccessfully
C:\Users\mboehme\github\forks\grpc-device\source\tests\system\nifgen_driver_api_tests.cpp(556): error: Value of: status.ok()
  Actual: false
Expected: true
[  FAILED  ] NiFgenDriverApiTest.SendSoftwareEdgeTrigger_TriggersSuccessfully (76 ms)
[ RUN      ] NiFgenDriverApiTest.AllocateNamedWaveform_WriteNamedWaveformF64_WaveformWrittenSuccessfully
C:\Users\mboehme\github\forks\grpc-device\source\tests\system\nifgen_driver_api_tests.cpp(352): error: Value of: status.ok()
  Actual: false
Expected: true
C:\Users\mboehme\github\forks\grpc-device\source\tests\system\nifgen_driver_api_tests.cpp(368): error: Value of: status.ok()
  Actual: false
Expected: true
[  FAILED  ] NiFgenDriverApiTest.AllocateNamedWaveform_WriteNamedWaveformF64_WaveformWrittenSuccessfully (94 ms)
[ RUN      ] NiFgenDriverApiTest.AllocateWaveform_WriteWaveformComplexF64_WaveformWrittenSuccessfully
C:\Users\mboehme\github\forks\grpc-device\source\tests\system\nifgen_driver_api_tests.cpp(383): error: Value of: status.ok()
  Actual: false
Expected: true
C:\Users\mboehme\github\forks\grpc-device\source\tests\system\nifgen_driver_api_tests.cpp(404): error: Value of: status.ok()
  Actual: false
Expected: true
[  FAILED  ] NiFgenDriverApiTest.AllocateWaveform_WriteWaveformComplexF64_WaveformWrittenSuccessfully (87 ms)
[ RUN      ] NiFgenDriverApiTest.OutputModeConfiguredToSeq_CreateAdvancedArbSequenceForArbitraryWaveform_CreatesSuccessfully
C:\Users\mboehme\github\forks\grpc-device\source\tests\system\nifgen_driver_api_tests.cpp(319): error: Value of: status.ok()
  Actual: false
Expected: true
C:\Users\mboehme\github\forks\grpc-device\source\tests\system\nifgen_driver_api_tests.cpp(336): error: Value of: status.ok()
  Actual: false
Expected: true
[  FAILED  ] NiFgenDriverApiTest.OutputModeConfiguredToSeq_CreateAdvancedArbSequenceForArbitraryWaveform_CreatesSuccessfully (106 ms)
[ RUN      ] NiFgenDriverApiTest.OutputModeConfiguredToArb_CreateWaveformI16_CreatesSuccessfully
C:\Users\mboehme\github\forks\grpc-device\source\tests\system\nifgen_driver_api_tests.cpp(636): error: Value of: status.ok()
  Actual: false
Expected: true
[  FAILED  ] NiFgenDriverApiTest.OutputModeConfiguredToArb_CreateWaveformI16_CreatesSuccessfully (47 ms)
```

Reply:
Update this change just to update the NiFgenSessionTests for now.

Reply:
I think the error reporting in the MI driver tests needs some improvements.

```
  ::grpc::Status status = GetStub()->SendSoftwareEdgeTrigger(&context, request, &response);

  EXPECT_TRUE(status.ok());
  expect_api_success(response.status());
```

`EXPECT_TRUE(status.ok());` reports whether the returned gRPC status is OK or error. This is correct, but not very descriptive. It doesn't tell you the gRPC status code (UNKNOWN, INVALID_ARGUMENT, etc.) or the error message.

`expect_api_success(response.status());` checks the NI status code in the response. However, NI gRPC Device Server no longer reports errors this way: it reports errors by returning a non-OK gRPC status. This means that error cases leave the response object's status field set to 0.

I updated [nidaqmx_driver_api_tests.cpp](https://github.com/ni/grpc-device/blob/a8dc635e02d146b999c3859dcadb1534efabfaa8/source/tests/system/nidaqmx_driver_api_tests.cpp) to use `client::raise_if_error()` for all or most API calls. 

This is more helpful because it:
1. ensures that tests don't ignore unexpected errors
2. prints the error message (and maybe the gRPC status code, I'm not sure)

Reply:
BTW, one of those tests is OSP-specific like @marcoskirsch mentioned

Reply:
I created #986 to handle updating those tests.

Reply:
Talked with Brad offline about maybe also disabling the NiFgenDriverApiTest. I think I would rather see if updating the NiFgenDriverApiTest to use the new hardware does in fact stop us from seeing the issue and see if we see more instances in NiFgenDriverApi test failing while looking into updating the tests.